File: Math/leetcode_67.cpp
// Problem Link ====>>https://leetcode.com/problems/add-binary/description/

#include<iostream>
    using namespace std;

class Solution
{
public:
    string addBinary(string a, string b)
    {
        string result = "";
        int carry = 0;
        int i = a.length() - 1;
        int j = b.length() - 1;
        while (i >= 0 || j >= 0)
        {
            int sum = carry;
            if (i >= 0)
            {
                sum += a[i] - '0';
            }
            if (j >= 0)
            {
                sum += b[j] - '0';
            }
            carry = sum / 2;
            result = char(sum % 2 + '0') + result;
            i--;
            j--;
        }
        if (carry == 1)
        {
            result = '1' + result;
        }
        return result;
    }
};

int main(){
    Solution s;
    string result = s.addBinary("11", "1");
    cout << result << endl; // prints 100

    return 0;
}

File: Math/leetcode_67.py
# Problem Link ====>>https://leetcode.com/problems/add-binary/description/

class Solution:
    def addBinary(self, a: str, b: str) -> str:
        """
        Adds two binary strings.

        Args:
            a: The first binary string.
            b: The second binary string.

        Returns:
            The sum of `a` and `b` in binary.
        """

        result = ""
        carry = 0
        i = len(a) - 1
        j = len(b) - 1
        while i >= 0 or j >= 0:
            sum = carry
            if i >= 0:
                sum += int(a[i])
            if j >= 0:
                sum += int(b[j])
            carry = sum // 2
            result = str(sum % 2) + result
            i -= 1
            j -= 1

        if carry == 1:
            result = "1" + result

        return result

solution = Solution()
result = solution.addBinary("11", "1")
print(result)  # prints 100
